test/store_unittest.cc

```diff
@@ -0,0 +1,226 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/time.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataStoreTest : public ::testing::Test {
+ protected:
+  Configuration config;
+  MetadataStore store;
+
+  MetadataStoreTest() : config(), store(config) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() {
+    return store.GetMetadataMap();
+  }
+
+  void PurgeDeletedEntries() {
+    store.PurgeDeletedEntries();
+  }
+
+};
+
+TEST_F(MetadataStoreTest, ResourceID) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id"}, mr);
+  EXPECT_EQ("some_resource", store.LookupResource("id").type());
+}
+
+TEST_F(MetadataStoreTest, ResourceNotFound) {
+  EXPECT_THROW(store.LookupResource("some_resource_id"), std::out_of_range);
+}
+
+TEST_F(MetadataStoreTest, ManyResources) {
```

Comment:
I wonder if we could come up with some consistent naming scheme, e.g., `OneResource` above, `MultipleResources` here, `OneResourceMultipleIds` below, then `MultipleResourcesMultipleIds`, etc...

Reply:
Done.